derivedmscal/DerivedMC/UDFMSCal.cc

```diff
@@ -453,8 +453,8 @@ namespace casacore {
         Vector<Int> selectedAnts1;
         Vector<Int> selectedAnts2;
         Matrix<Int> selectedBaselines;
-        CountedPtr<MSSelectionErrorHandler> curHandler = MSAntennaParse::thisMSAErrorHandler;
-        MSAntennaParse::thisMSAErrorHandler = new UDFMSCalErrorHandler();
+        std::shared_ptr<MSSelectionErrorHandler> curHandler = MSAntennaParse::thisMSAErrorHandler;
+        MSAntennaParse::thisMSAErrorHandler.reset (new UDFMSCalErrorHandler());
```

Comment:
Using `std::move` avoids copying the shared pointer and adjusting the reference count, which requires locking an internal mutex.
```suggestion
        std::shared_ptr<MSSelectionErrorHandler> curHandler = std::move(MSAntennaParse::thisMSAErrorHandler);
        MSAntennaParse::thisMSAErrorHandler = std::make_shared<UDFMSCalErrorHandler>();
```

Reply:
I'll use swap to exchange the handlers.